src/main/interface/cli.c

```diff
@@ -970,6 +1003,16 @@ static void cliSerialPassthrough(char *cmdline)
                    passThroughPort->mode, mode);
             serialSetMode(passThroughPort, mode);
         }
+
+        // Set the baud rate if specified otherwise allow to be set via USB
+        if (baud) {
+            serialSetBaudRate(passThroughPort, baud);
+        } else {
+			// Register the right side baud rate setting routine with the left side which allows setting of the UART
+			// baud rate over USB without setting it using the serialpassthrough command
+			serialSetBaudRateCb(cliPort, serialSetBaudRate, passThroughPort);
```

Comment:
This may alter baudrate asynchronously to serial transmission, possibly corrupting data
(Actually not a big problem in practice)

Reply:
Indeed, this is a possibility, but this is typically done when a port is opened, before data transmission, not during, and whilst the serial data could be queued in a buffer, the baud rate change isn't buffered. So, trying to change the baud rate after sending some serial data would be dangerous, but doing so beforehand there won't be a race condition.